sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java

```diff
@@ -39,6 +39,9 @@
 
     public FeedRangeEpkImpl(final Range<String> range) {
         checkNotNull(range, "Argument 'range' must not be null");
+        if (range.getMin().compareTo(range.getMax()) > 0) {
+            throw new IllegalArgumentException("The provided range is incorrect min is larger than max");
```

Comment:
To be on the safer side, I would suggest using `compareToIgnoreCase` - as I am not sure if backend would always return range in upper case. Just a thought.

Reply:
I checked the code again, at other places, we are using `compareTo()`, so I guess we will keep it the same here for consistency, doesn't make sense to introduce any weird behavior.

Reply:
Yes - I actually checked that we will always use upper case intentionally to not have to use any ignore case comparisons. The "string" is always HEX for the range (at least in reality - why the class was made a Generic in the first place when it is only ever being used as Range<String> is a different question - it spans .Net and Java as well)